src/frontend/screens/Sync/ProjectSyncDisplay.tsx

```diff
@@ -99,146 +184,344 @@ export const ProjectSyncDisplay = ({
         </View>
       </View>
-      <Text style={styles.titleText}>{devicesSyncingText}</Text>
-      {!isSyncDone && isDataSyncEnabled && <SyncProgress />}
+      {syncInfoContent}
     </ScreenContentWithDock>
   );
 };
 
-function SyncProgress() {
+function SyncProgress({
+  stage,
+}: {
+  stage: Extract<
+    SyncStage,
+    {name: 'syncing' | 'waiting' | 'complete-partial' | 'complete-full'}
+  >;
+}) {
   const {formatMessage: t} = useIntl();
 
-  const progress = useSyncProgress();
+  let progressLabel: string;
 
-  const dynamicProgressBarProps =
-    progress === null
-      ? {indeterminate: true, indeterminateAnimationDuration: 2000}
-      : {
-          progress,
-          indeterminate: false,
-        };
+  switch (stage.name) {
+    case 'waiting': {
+      progressLabel = t(m.progressLabelWaiting);
+      break;
+    }
+    case 'syncing': {
+      progressLabel = t(m.progressLabelSyncing);
+      break;
+    }
+    case 'complete-partial': {
+      progressLabel = t(m.progressLabelWithDeviceCount, {
+        active: stage.syncingPeersCount,
+        total: stage.connectedPeersCount,
+      });
+      break;
+    }
+    case 'complete-full': {
+      progressLabel = t(m.progressLabelComplete, {
+        count: stage.connectedPeersCount,
+      });
+      break;
+    }
+    default: {
+      // @ts-expect-error
+      throw new Error(`Unknown sync status ${stage.name}`);
+    }
+  }
 
   return (
     <View style={styles.syncProgressContainer}>
       <View style={styles.syncProgressTextContainer}>
-        <SyncIcon color={COMAPEO_BLUE} size={20} />
-        <Text style={styles.syncProgressTitleText}>{t(m.syncing)}</Text>
+        {stage.name === 'complete-full' ? (
+          <DoneIcon color={DARK_GREEN} size={20} />
+        ) : (
+          <SyncIcon color={COMAPEO_BLUE} size={20} />
+        )}
+        <Text
+          style={[
+            styles.syncProgressLabel,
+            stage.name === 'complete-full' && {color: DARK_GREEN},
+          ]}>
+          {progressLabel}
+        </Text>
       </View>
       <ProgressBar
-        {...dynamicProgressBarProps}
+        {...(stage.name === 'waiting'
+          ? {indeterminate: true, indeterminateAnimationDuration: 2000}
+          : {progress: stage.progress, indeterminate: false})}
         height={10}
         width={null}
         borderRadius={0}
-        color={COMAPEO_BLUE}
+        color={stage.name === 'complete-full' ? DARK_GREEN : COMAPEO_BLUE}
         unfilledColor={LIGHT_GREY}
         borderColor={WHITE}
       />
 
-      {progress !== null && (
+      {stage.name !== 'waiting' && (
         <Text style={styles.syncProgressText}>
-          {t(m.syncProgress, {value: Math.round(progress * 100)})}
+          {t(m.progressSyncPercentage, {
+            value: Math.round(stage.progress * 100),
+          })}
         </Text>
       )}
     </View>
   );
 }
 
+function deriveSyncStage({
```

Comment:
Seems like this could be useful to test? Wrote tests in [this gist](https://gist.github.com/EvanHahn/c3fd5f693a9dd2ac2f27d3daa8f31413).

Reply:
addressed via https://github.com/digidem/comapeo-mobile/pull/619/commits/cccd6df1d956af886691cb72de90ac1eb98873f2